include/alpaka/atomic/AtomicUniformCudaHipBuiltIn.hpp

```diff
@@ -177,22 +177,51 @@ namespace alpaka
         {
             //
             __device__ static auto atomicOp(
-                AtomicUniformCudaHipBuiltIn const&,
+                AtomicUniformCudaHipBuiltIn const& ctx,
                 [[maybe_unused]] unsigned long int* const addr,
                 [[maybe_unused]] unsigned long int const& value) -> unsigned long int
             {
-                // LLP64
-                if constexpr(std::numeric_limits<unsigned int>::max() == std::numeric_limits<unsigned long>::max())
-                    return ::atomicSub(reinterpret_cast<unsigned int*>(addr), static_cast<unsigned int>(value));
-                else // LP64
-                    static_assert(
-                        !sizeof(THierarchy),
-                        "atomicOp<AtomicSub, AtomicUniformCudaHipBuiltIn, unsigned long int> is only supported when "
-                        "sizeof(unsigned long int) == 4");
+                return atomicAdd(ctx, addr, -value); // Calls alpaka::atomicAdd function
```

Comment:
```suggestion
                return alpaka::atomicAdd(ctx, addr, -value);
```
If you want to insist on which function is called, you can also spell it out then :)